File: mobile/Android/YoloTinyV1Tensorflow/app/src/main/java/io/pry/ml/yolo/tiny/v1/tensorflow/utils/ColorPalette.kt
package io.pry.ml.yolo.tiny.v1.tensorflow.utils

import android.graphics.Color
import android.graphics.Paint


//https://sashat.me/2017/01/11/list-of-20-simple-distinct-colors/
object ColorPalette {

    //region defs
    private val colors = arrayOf(
        Color.argb(255, 230, 25, 75),
        Color.argb(255, 60, 180, 75),
        Color.argb(255, 255, 225, 25),
        Color.argb(255, 0, 130, 200),
        Color.argb(255, 245, 130, 48),
        Color.argb(255, 145, 30, 180),
        Color.argb(255, 70, 240, 240),
        Color.argb(255, 240, 50, 230),
        Color.argb(255, 210, 245, 60),
        Color.argb(255, 250, 190, 190),
        Color.argb(255, 0, 128, 128),
        Color.argb(255, 230, 190, 255),
        Color.argb(255, 170, 110, 40),
        Color.argb(255, 255, 250, 200),
        Color.argb(255, 128, 0, 0),
        Color.argb(255, 170, 255, 195),
        Color.argb(255, 128, 128, 0),
        Color.argb(255, 255, 215, 180),
        Color.argb(255, 0, 0, 128),
        Color.argb(255, 128, 128, 128),
        Color.argb(255, 255, 255, 255),
        Color.argb(255, 0, 0, 0)
    )
    //endregion

    //region state
    private val paints: Array<Paint> = Array(colors.size, { i-> createPaint(8f, colors[i]) })
    //endregion


    //region public
    fun paintForIndex(i: Int): Paint {
        return paints[i % paints.size]
    }
    //endregion`

    //region helpers
    private fun createPaint(strokeWidth: Float, color: Int): Paint {
        val paint = Paint()
        paint.color = color
        paint.strokeWidth = strokeWidth
        paint.style = Paint.Style.STROKE
        paint.textSize = 48f
        return paint
    }
    //endregion
}

File: mobile/iOS/YoloTinyV1Tensorflow/YoloTinyV1Tensorflow/Utils/ColorPalette.swift
import Foundation

//https://sashat.me/2017/01/11/list-of-20-simple-distinct-colors/
class ColorPalette {
    
    private static let colors = [
        color(red: 60, green: 180, blue: 75),
        color(red: 230, green: 25, blue: 75),
        color(red: 255, green: 225, blue: 25),
        color(red: 0, green: 130, blue: 200),
        color(red: 245, green: 130, blue: 48),
        color(red: 145, green: 30, blue: 180),
        color(red: 70, green: 240, blue: 240),
        color(red: 240, green: 50, blue: 230),
        color(red: 210, green: 245, blue: 60),
        color(red: 250, green: 190, blue: 190),
        color(red: 0, green: 128, blue: 128),
        color(red: 230, green: 190, blue: 255),
        color(red: 170, green: 110, blue: 40),
        color(red: 255, green: 250, blue: 200),
        color(red: 128, green: 0, blue: 0),
        color(red: 170, green: 255, blue: 195),
        color(red: 128, green: 128, blue: 0),
        color(red: 255, green: 215, blue: 180),
        color(red: 0, green: 0, blue: 128),
        color(red: 128, green: 128, blue: 128),
        color(red: 255, green: 255, blue: 255),
        color(red: 0, green: 0, blue: 0)
    ]
    
    static func colorForIndex(i: Int) -> UIColor {
        return colors[i % colors.count]
    }
    
    private static func color(red: CGFloat, green: CGFloat, blue: CGFloat) -> UIColor {
        return UIColor(red: red / 255, green: green / 255, blue: blue / 255, alpha: 1)
    }
}
